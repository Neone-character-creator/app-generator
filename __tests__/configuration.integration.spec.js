const fs = require("fs-extra");
const path = require("path");

describe("plugin generation chain", () => {
   it("completes", async done => {
       try {
           const configFile = await require("../lib/configurationSchema")(JSON.parse(fs.readFileSync(path.resolve(__dirname, `test.json`), 'utf-8')));
           const hierarchy = require('../lib/generateComponentsHierarchy').default(configFile);
           const components = await require('../lib/componentGenerator')(hierarchy);
           expect(components.views[0]).toBe("summary");
           expect(components.components.summary).toEqual({
               type: "view",
               children: [
                   "biography",
                   "combat",
                   "characteristics",
                   "skills"
               ]
           });

           expect(component.components.biography).toEqual({
               type: "container",
               direction: "vertical",
               children: [
                   "name",
                   "species",
                   "career",
                   "specializations"
               ]
           })
       } catch (e) {
           done(e);
       }
   });
});