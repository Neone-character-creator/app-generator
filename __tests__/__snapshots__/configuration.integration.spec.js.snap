// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin generation chain completes 1`] = `
"import React from \\"react\\";
import * as _ from \\"lodash\\";
import models from \\"../models\\";
import interpreter from \\"../interpreter\\";
import Checkbox from \\"@material-ui/core/Checkbox\\";
import Textfield from \\"@material-ui/core/TextField\\";
import FormControl from \\"@material-ui/core/FormControl\\";
import InputLabel from \\"@material-ui/core/InputLabel\\";
import Select from \\"@material-ui/core/Select\\";
import MenuItem from \\"@material-ui/core/MenuItem\\";
import List from \\"@material-ui/core/List\\";
import ListItem from \\"@material-ui/core/ListItem\\";
import Grid from \\"@material-ui/core/Grid\\";
import Button from \\"@material-ui/core/Button\\";
import BiographyContainer from './BiographyContainer';
import CombatContainer from './CombatContainer';
import CharacteristicsContainer from './CharacteristicsContainer';
import SkillsContainer from './SkillsContainer';
import WeaponsContainer from './WeaponsContainer';
import ArmorContainer from './ArmorContainer';
import ExperienceContainer from './ExperienceContainer';
import MotivationsContainer from './MotivationsContainer';

const Component = () => (
    <div id=\\"Summary-view\\">
        <BiographyContainer />
<CombatContainer />
<CharacteristicsContainer />
<SkillsContainer />
<WeaponsContainer />
<ArmorContainer />
<ExperienceContainer />
<MotivationsContainer />
    </div>
)
export default Component;"
`;
